lib/manager/npm/post-update/index.ts

```diff
@@ -434,24 +434,20 @@ export async function getAdditionalFiles(
 
   const { additionalNpmrcContent, additionalYarnRcYml } = processHostRules();
 
-  const env = getChildProcessEnv([
-    'NPM_CONFIG_CACHE',
-    'YARN_CACHE_FOLDER',
-    'npm_config_store',
-  ]);
-  env.NPM_CONFIG_CACHE =
-    env.NPM_CONFIG_CACHE ||
-    upath.join(getAdminConfig().cacheDir, './others/npm');
-  await ensureDir(env.NPM_CONFIG_CACHE);
-  env.YARN_CACHE_FOLDER =
-    env.YARN_CACHE_FOLDER ||
-    upath.join(getAdminConfig().cacheDir, './others/yarn');
-  await ensureDir(env.YARN_CACHE_FOLDER);
-  env.npm_config_store =
-    env.npm_config_store ||
-    upath.join(getAdminConfig().cacheDir, './others/pnpm');
-  await ensureDir(env.npm_config_store);
-  env.NODE_ENV = 'dev';
+  const env = {
+    ...getChildProcessEnv([
+      'NPM_CONFIG_CACHE',
+      'YARN_CACHE_FOLDER',
+      'npm_config_store',
+    ]),
+    NPM_CONFIG_CACHE: await ensureCacheDir('./others/npm', 'NPM_CONFIG_CACHE'),
```

Comment:
I find this confusing compared to the above line where these variables are passed as a command to `getChildProcessEnv`. Is it a mistake or can you explain to me what the flow is?

Reply:
Ahh, is it because `ensureCacheDir()` will return the value of the env variable if it's already set?

Reply:
Never mind, I've simplified it